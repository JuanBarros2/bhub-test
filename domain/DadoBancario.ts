export type DadoBancario = {
    agencia: string;
    conta: string;
    banco: string;
}