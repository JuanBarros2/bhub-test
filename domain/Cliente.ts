import { DadoBancario } from './DadoBancario';

export type ClientId = string;

export type Cliente = {
  id: ClientId;
  razaoSocial: string;
  telefone: string;
  endereco: string;
  dataCadastro: string;
  faturamentoDeclarado: number;
  dadosBancarios: DadoBancario[]
}