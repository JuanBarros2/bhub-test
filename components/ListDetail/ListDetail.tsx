import * as React from "react";
import { Cliente } from "../../domain/Cliente";
import DeleteClienteButton from "../DeleteClienteButton/DeleteClienteButton";
import UpdateClienteForm from "../UpdateClienteForm/UpdateClienteForm";

type ListDetailProps = {
  cliente: Cliente;
};

const ListDetail = ({ cliente }: ListDetailProps) => (
  <div className="flex flex-row gap-10 justify-center">
    <div>
      <h2 className="text-lg">{cliente.razaoSocial}</h2>
      <div className="flex flex-col mb-8">
        <span>Telefone: {cliente.telefone}</span>
        <span>Endereço: {cliente.endereco}</span>
        <span>Data de Criação: {cliente.dataCadastro.toDateString()}</span>
        <span>Faturamento declarado: {cliente.faturamentoDeclarado}</span>
      </div>
      <div className="flex flex-row gap-2">
        <DeleteClienteButton cliente={cliente} />
      </div>
    </div>
    <div>
      <UpdateClienteForm cliente={cliente} />
    </div>
  </div>
);

export default ListDetail;
