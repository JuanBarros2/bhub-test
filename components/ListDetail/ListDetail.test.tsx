import { render } from "@testing-library/react";
import { Cliente } from "../../domain/Cliente";
import ListDetail from "./ListDetail";

const mockedCliente: Cliente = {
  id: "1",
  dadosBancarios: [],
  dataCadastro: new Date("28/02/2012"),
  endereco: "Rua Nova",
  faturamentoDeclarado: 1111,
  razaoSocial: "Bhub",
  telefone: "99999999",
};

describe("Home", () => {
  it("should match with snapshot", () => {
    const { container } = render(<ListDetail cliente={mockedCliente} />);
    expect(container).toMatchSnapshot();
  });
});
