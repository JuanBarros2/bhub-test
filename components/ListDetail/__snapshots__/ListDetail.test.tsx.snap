// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match with snapshot 1`] = `
<div>
  <div
    class="flex flex-row gap-10 justify-center"
  >
    <div>
      <h2
        class="text-lg"
      >
        Bhub
      </h2>
      <div
        class="flex flex-col mb-8"
      >
        <span>
          Telefone: 
          99999999
        </span>
        <span>
          Endereço: 
          Rua Nova
        </span>
        <span>
          Data de Criação: 
          Invalid Date
        </span>
        <span>
          Faturamento declarado: 
          1111
        </span>
      </div>
      <div
        class="flex flex-row gap-2"
      >
        <button
          class="btn btn-secondary"
        >
          Deletar
        </button>
      </div>
    </div>
    <div>
      <div>
        <form
          class="flex flex-col w-fit justify-center"
        >
          <input
            placeholder="Razão Social"
            required=""
            type="text"
          />
          <input
            placeholder="Telefone"
            required=""
            type="text"
          />
          <input
            placeholder="Endereço"
            required=""
            type="text"
          />
          <input
            placeholder="Faturamento"
            required=""
            type="text"
          />
          <input
            class="btn btn-primary"
            type="submit"
            value="Editar cliente"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
